pkg/minikube/update/update.go

```diff
@@ -175,27 +190,54 @@ func updateBinary(v semver.Version, downloadBinary, updateLinkPrefix, downloadLi
 }
 
 func updateBinaryFile(url string, checksum []byte) {
-	fmt.Println("Downloading updated binary")
+	var binary string
+	fmt.Println("Downloading updated archive")
 	httpResp, err := http.Get(url)
 	if err != nil {
-		glog.Errorf("Cannot download binary: %s", err)
+		glog.Errorf("Cannot download archive: %s", err)
 		os.Exit(1)
 	}
 	defer func() { _ = httpResp.Body.Close() }()
 
-	binary := httpResp.Body
+	updatedArchive := httpResp.Body
 	if httpResp.ContentLength > 0 {
 		bar := pb.New64(httpResp.ContentLength).SetUnits(pb.U_BYTES)
 		bar.Start()
-		binary = bar.NewProxyReader(binary)
+		updatedArchive = bar.NewProxyReader(updatedArchive)
 		defer func() {
 			<-time.After(bar.RefreshRate)
 			fmt.Println()
 		}()
 	}
-	err = update.Apply(binary, update.Options{
-		Hash:     crypto.SHA256,
-		Checksum: checksum,
+
+	// verify checksum
+	archiveBytes, err := ioutil.ReadAll(updatedArchive)
+
+	archiveChecksum, err := checksumFor(crypto.SHA256, archiveBytes)
+
+	if !bytes.Equal(archiveChecksum, checksum) {
+		fmt.Println(fmt.Errorf("Updated file has wrong checksum. Expected: %x, got: %x", archiveChecksum, checksum))
+		os.Exit(1)
+	}
+
+	// Create a temporary directory to store archive contents
+	dir, err := ioutil.TempDir("", "minishift")
+	if err != nil {
+		atexit.ExitWithMessage(1, "Could not create a temporary directory to store archive contents")
```

Comment:
We should not be using atexit anywhere outside of the commands. If there is an error, it should be returned.